metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,413 @@
+CASSANDRA_SYSTEM_SCHEMA_TABLES = {
+    "keyspaces": "keyspaces",
+    "tables": "tables",
+    "columns": "columns",
+}
+# these column names are present on the system_schema tables
+CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES = {
+    "keyspace_name": "keyspace_name",
+    "table_name": "table_name",
+    "column_name": "column_name",
+    "column_type": "type",
+}
+
+# -------------------------------------------------- queries -------------------------------------------------- #
+# get all keyspaces
+GET_KEYSPACES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['keyspaces']}"
+# get all tables for a keyspace
+GET_TABLES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['tables']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s"
+# get all columns for a table
+GET_COLUMNS_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['columns']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s AND {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['table_name']} = %s"
+
+# -------------------------------------------------- setup the config and source reporter -------------------------------------------------- #
+
+# config
+class CassandraSourceConfig(PlatformInstanceConfigMixin, EnvConfigMixin):
+    contact_point: str = Field(
+        default="localhost", description="The cassandra instance contact point domain (without the port)."
+    )
+    port: str = Field(
+        default="10350", description="The cassandra instance port."
+    )
+    username: str = Field(
+        default=None, description="The username credential."
+    )
+    password: str = Field(
+        default=None, description="The password credential."
+    )
+    excludeKeyspaces: List[str] = Field(
+        default=[], description="The keyspaces to exclude."
+    )
+
+
+# source reporter
+@dataclass
+class CassandraSourceReport(SourceReport):
+    filtered: List[str] = field(default_factory=list)
+
+    def report_dropped(self, index: str) -> None:
+        self.filtered.append(index)
+
+
+
+# -------------------------------------------------- the main source class -------------------------------------------------- #
+@platform_name("Cassandra")
+@config_class(CassandraSourceConfig)
+@support_status(SupportStatus.TESTING)
+@capability(SourceCapability.PLATFORM_INSTANCE, "Enabled by default")
+class CassandraSource(Source):
+
+    """
+    This plugin extracts the following:
+
+    - Metadata for tables
+    - Column types associated with each table column
+    - The keyspace each table belongs to
+    """
+
+    config: CassandraSourceConfig
+    report: CassandraSourceReport
+    cassandra_session: Session
+
+
+    def __init__(self, ctx: PipelineContext, config: CassandraSourceConfig):
+        self.ctx = ctx
+        self.config = config
+        self.report = CassandraSourceReport()
+
+        # attempt to connect to cass
```

Comment:
I noticed that it's possible to allow for local file paths to cert files (ie if they're using cert based auth), but the UI doesn't support it. For this reason I kept it to just username/password. But correct me if I'm wrong and happy to add an option in the config for cert auth. Especially if we think this would be a big value add for customers.